pkg/sql/executor.go

```diff
@@ -446,15 +446,21 @@ func (e *Executor) Prepare(
 
 	// Prepare needs a transaction because it needs to retrieve db/table
 	// descriptors for type checking.
-	// TODO(andrei): is this OK? If we're preparing as part of a SQL txn, how do
-	// we check that they're reading descriptors consistent with the txn in which
-	// they'll be used?
-	txn := client.NewTxn(e.cfg.DB)
-	if err := txn.SetIsolation(session.DefaultIsolationLevel); err != nil {
-		panic(err)
+	txn := session.TxnState.txn
+	if txn == nil {
+		// The new txn need not be the same transaction used by statements following
```

Comment:
If it doesn't need to be the same, then why ever make it the same? Why is this change necessary exactly?

I think that this is the opposite of what we generally want to do; I think we want to always obtain leases in a fresh transaction - otherwise what's preventing us from obtaining a lease at an arbitrarily old version of a descriptor? (i.e. a client txn that's been lingering for a while might not see new versions of the descriptor... unless its attempt to write a lease at an old version is guaranteed to conflict with reads that will have been performed in the process of incrementing the descriptor version...).
There's some subtleties here I'm not currently sure of.
If you don't mind, I'd like to talk to you about our lease philosophy more. But I'm just boarding the plane for this California meetup.